### PR TITLE
lib: verbose output during installation

### PR DESCRIPTION
**Pull request by pinheadmz**
Replaces `execFile` with `spawn` to enable output logging.
Adds console logging for main functions.

Overall provides some progress to watch.

Example:

```
--> gpm install
Installing: /Users/matthewzipkin/Desktop/work/gpm-bcoin/bcoin
Cloning: https://github.com/bcoin-org/bcfg.git v0.1.6
Cloning: https://github.com/bcoin-org/bcrypto.git v3.1.11
Cloning: https://github.com/bcoin-org/bcurl.git v0.1.6
Cloning: https://github.com/bcoin-org/bdb.git v1.1.7
Cloning: https://github.com/bcoin-org/bdns.git v0.1.5
Cloning: https://github.com/bcoin-org/bevent.git v0.1.5
Cloning: https://github.com/bcoin-org/bfile.git v0.2.1
Cloning: https://github.com/bcoin-org/bfilter.git v1.0.5
Cloning: https://github.com/bcoin-org/bheep.git v0.1.5
Cloning: https://github.com/bcoin-org/binet.git v0.3.5
Cloning: https://github.com/bcoin-org/blgr.git v0.1.7
Cloning: https://github.com/bcoin-org/blru.git v0.1.6
Cloning: https://github.com/bcoin-org/blst.git v0.1.5
Cloning: https://github.com/bcoin-org/bmutex.git v0.1.6
Cloning: https://github.com/bcoin-org/brq.git v0.1.8
Cloning: https://github.com/bcoin-org/bs32.git v0.1.6
Cloning: https://github.com/chjj/bsert.git v0.0.10
Cloning: https://github.com/bcoin-org/bsip.git v0.1.9
Cloning: https://github.com/bcoin-org/bsock.git v0.1.9
Cloning: https://github.com/bcoin-org/bsocks.git v0.2.5
Cloning: https://github.com/bcoin-org/bstring.git v0.3.9
Cloning: https://github.com/bcoin-org/btcp.git v0.1.5
Cloning: https://github.com/chjj/buffer-map.git v0.0.7
Cloning: https://github.com/bcoin-org/bufio.git v1.0.6
Cloning: https://github.com/bcoin-org/bupnp.git v0.2.6
Cloning: https://github.com/bcoin-org/bval.git v0.1.6
Cloning: https://github.com/bcoin-org/bweb.git v0.1.9
Cloning: https://github.com/chjj/loady.git v0.0.1
Cloning: https://github.com/bcoin-org/mrmr.git v0.1.8
Cloning: https://github.com/chjj/n64.git v0.2.10
Cloning: https://github.com/braydonf/nan.git v2.13.2
Cloning: https://github.com/bcoin-org/bmocha.git v2.1.3
Installing: /Users/matthewzipkin/Desktop/work/gpm-bcoin/bcoin/node_modules/bcfg
Installing: /Users/matthewzipkin/Desktop/work/gpm-bcoin/bcoin/node_modules/bcrypto
Building: /Users/matthewzipkin/Desktop/work/gpm-bcoin/bcoin/node_modules/bcrypto
  CC(target) Release/obj.target/bcrypto/src/aead/aead.o

  CC(target) Release/obj.target/bcrypto/src/aes/aes.o

  CC(target) Release/obj.target/bcrypto/src/blake2b/blake2b.o

  CC(target) Release/obj.target/bcrypto/src/blake2s/blake2s.o

  CC(target) Release/obj.target/bcrypto/src/chacha20/chacha20.o

  CC(target) Release/obj.target/bcrypto/src/dsa/dsa.o

...
```

### CODE REVIEW

**Comment by pinheadmz** [comment]
We can watch the signature checks too:

```diff
diff --git a/lib/git.js b/lib/git.js
index 6ec52fb..081add3 100644
--- a/lib/git.js
+++ b/lib/git.js
@@ -75,7 +75,7 @@ async function verifyRepo(tag, commit, dst) {
     result = await exec(`git verify-tag ${tag}`, {cwd: dst});
   else
     result = await exec(`git verify-commit ${commit}`, {cwd: dst});
-
+  console.log(result.stderr)
   return true;
 }

```

### Example
```
--> gpm install
Installing: /Users/matthewzipkin/Desktop/work/gpm-bcoin/bcoin
Cloning: https://github.com/bcoin-org/bcfg.git v0.1.6
gpg: Signature made Thu Mar 21 15:07:59 2019 EDT
gpg:                using RSA key B4B1F62DBAC084E333F3A04A8962AB9DE6666BBD
gpg: Good signature from "Christopher Jeffrey <chjjeffrey@gmail.com>" [full]

Cloning: https://github.com/bcoin-org/bcrypto.git v3.1.11
gpg: Signature made Wed May  8 15:01:16 2019 EDT
gpg:                using RSA key B4B1F62DBAC084E333F3A04A8962AB9DE6666BBD
gpg: Good signature from "Christopher Jeffrey <chjjeffrey@gmail.com>" [full]

Cloning: https://github.com/bcoin-org/bcurl.git v0.1.6
gpg: Signature made Thu Mar 21 15:08:22 2019 EDT
gpg:                using RSA key B4B1F62DBAC084E333F3A04A8962AB9DE6666BBD
gpg: Good signature from "Christopher Jeffrey <chjjeffrey@gmail.com>" [full]
```

**Comment by tuxcanfly** [comment]
Hmm, we could add verbose levels to fine tune how detailed we want these.

File: lib/gpm.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ const path = require('path');
 const semver = require('../vendor/semver');
 const readFile = util.promisify(fs.readFile);
 const access = util.promisify(fs.access);
-const {execFile} = child_process;
+const {execFile, spawn} = child_process;
```
Comment (boymanjor) [inline lib/gpm.js:26]: @pinheadmz if `execFile` is not used anymore, we should remove it from this import statement.

**Comment by braydonf** [comment]
Logging has been added.

**Pull request closed without merging**